### PR TITLE
feat: Photo storage example

### PR DESCRIPTION
**Pull request by sea-snake**
Example React application that shows how to upload files to an assets canister with the `@dfinity/assets` package.

Related PR https://github.com/dfinity/agent-js/pull/603

### CODE REVIEW

**Comment by sesi200** [comment]
Thank you very much for this PR! I'd like to get back to this once https://github.com/dfinity/agent-js/pull/603 is merged. Feel free to ping me when you think I should take another look

File: hosting/photo-storage/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+This example project can be cloned, installed and deployed locally, for learning and testing purposes. The instructions are based on running the example on either macOS or Linux, but when using WSL2 on Windows, the instructions will be the same.
+
+### Prerequisites
+The example project requires the following installed:
+
+- git
+- dfx
+- npm
+
+git can be installed from various package managers. DFX can be installed following the instructions [here](https://internetcomputer.org/docs/current/developer-docs/build/install-upgrade-remove).
```
Comment (sesi200) [inline hosting/photo-storage/README.md:16]:
```suggestion
git and npm can be installed from various package managers. DFX can be installed following the instructions [here](https://internetcomputer.org/docs/current/developer-docs/build/install-upgrade-remove).
```

File: hosting/photo-storage/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+$ dfx deploy
+```
+
+## Authorization
+To authorize the identity from this example project on a local network to upload files, it must be authorized first:
+```bash
+dfx canister call photo-storage authorize '(principal "535yc-uxytb-gfk7h-tny7p-vjkoe-i4krp-3qmcl-uqfgr-cpgej-yqtjq-rqe")'
+```
+
+Before deployment on the IC, the hardcoded identity should be replaced by an authentication method e.g. Internet Identity.
```
Comment (sesi200) [inline hosting/photo-storage/README.md:53]:
```suggestion
Before deployment on the IC, the hardcoded identity (defined in `src/App.js`) should be replaced by an authentication method e.g. Internet Identity.
```

File: hosting/photo-storage/package.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "photo-storage",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@dfinity/assets": "^0.13.4",
+    "@dfinity/agent": "^0.13.4",
+    "@dfinity/identity": "^0.13.4",
```
Comment (sesi200) [inline hosting/photo-storage/package.json:8]:
```suggestion
    "@dfinity/assets": "^0.14.0",
    "@dfinity/agent": "^0.14.0",
    "@dfinity/identity": "^0.14.0",
    "@dfinity/candid": "^0.14.0",
    "@dfinity/principal": "^0.14.0",
```
I had to do this to get it to work

File: hosting/photo-storage/README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+```bash
+$ dfx deploy
+```
+
+## Authorization
+To authorize the identity from this example project on a local network to upload files, it must be authorized first:
+```bash
+dfx canister call photo-storage authorize '(principal "535yc-uxytb-gfk7h-tny7p-vjkoe-i4krp-3qmcl-uqfgr-cpgej-yqtjq-rqe")'
+```
```
Comment (sesi200) [inline hosting/photo-storage/README.md:51]: If you could add some sort of feedback if this gets forgotten that would be great! Even a simple `alert("please follow instructions in ##Authorization")` (or whatever the syntax in JS is for that) would be very helpful.

**Review by sesi200 — CHANGES_REQUESTED** [review]

**Comment by sea-snake** [comment]
Updated MR with above suggestions and fixed some bugs (replaced masonry dep with a more reliable masonry dep)

**Comment by sesi200** [comment]
Also, the `rust-basic-bitcoin` failed test will be fixed in master in a few minutes

**Comment by sea-snake** [comment]
> Looking nice, thanks a lot! Any chance you can also add a very basic e2e test to alert us (me) when a dfx update breaks the example? [This](https://github.com/dfinity/examples/blob/master/.github/workflows/rust-basic-bitcoin-example.yml) is a good example you could work from. It doesn't need to be anything fancy

Added a workflow that checks if the project builds and deploys successfully.

**Comment by sesi200** [comment]
With `npm i` and `npm run build` I get the following error:
```
Failed to compile.

Module not found: Error: Can't resolve 'assert' in '/Users/ssiff/Desktop/examples/hosting/photo-storage/node_modules/@dfinity/identity/lib/esm/identity'
BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "assert": require.resolve("assert/") }'
	- install 'assert'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "assert": false }
```

Is this a problem introduced by myself or something wrong with the code?

**Comment by sea-snake** [comment]
> With `npm i` and `npm run build` I get the following error:
> 
> ```
> Failed to compile.
> 
> Module not found: Error: Can't resolve 'assert' in '/Users/ssiff/Desktop/examples/hosting/photo-storage/node_modules/@dfinity/identity/lib/esm/identity'
> BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
> This is no longer the case. Verify if you need this module and configure a polyfill for it.
> 
> If you want to include a polyfill, you need to:
> 	- add a fallback 'resolve.fallback: { "assert": require.resolve("assert/") }'
> 	- install 'assert'
> If you don't want to include a polyfill, you can use an empty module like this:
> 	resolve.fallback: { "assert": false }
> ```
> 
> Is this a problem introduced by myself or something wrong with the code?

Seems like a problem in `@dfinity/identity 0.14.1`, updating dependencies to `0.15.0` seems to fix it.

**Comment by sesi200** [comment]
Thank you for fixing it that quickly! Can confirm it work now. Thank you very much for contributing!

**Pull request merged**